SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/ReferenceCountedLockingPolicy.cpp

```diff
@@ -118,24 +124,32 @@ bool UReferenceCountedLockingPolicy::IsLocked(const AActor* Actor) const
 		UE_LOG(LogReferenceCountedLockingPolicy, Warning, TEXT("IsLocked called for nullptr"));
 		return false;
 	}
-	return ActorToLockingState.Contains(Actor);
-}
 
-void UReferenceCountedLockingPolicy::OnLockedActorDeleted(AActor* DestroyedActor)
-{
-	TArray<ActorLockToken> TokensToRemove;
-	for (const auto& KeyValuePair : TokenToNameAndActor)
+	// Is this Actor explicitly locked or on a locked hierarchy ownership path.
+	if (IsExplicitlyLocked(Actor) || IsOnLockedHierarchyPath(Actor))
 	{
-		if (KeyValuePair.Value.Actor == DestroyedActor)
-		{
-			TokensToRemove.Add(KeyValuePair.Key);
-		}
+		return true;
 	}
-	for (const auto& Token : TokensToRemove)
+
+	// Is the hierarchy root of this Actor explicitly locked or on a locked hierarchy ownership path.
+	TArray<AActor*> OwnershipHierarchyPath = SpatialGDK::GetOwnershipHierarchyPath(Actor);
+	if (OwnershipHierarchyPath.Num() > 0)
 	{
-		TokenToNameAndActor.Remove(Token);
+		AActor* HierarchyRoot = OwnershipHierarchyPath[0];
+		return IsExplicitlyLocked(HierarchyRoot) || IsOnLockedHierarchyPath(HierarchyRoot);
 	}
-	ActorToLockingState.Remove(DestroyedActor);
+
+	return false;
+}
+
+bool UReferenceCountedLockingPolicy::IsExplicitlyLocked(const AActor* Actor) const
+{
+	return ActorToLockingState.Contains(Actor);
+}
+
+bool UReferenceCountedLockingPolicy::IsOnLockedHierarchyPath(const AActor* Actor) const
```

Comment:
I'm not a huge fan of reusing `path` here, as all Unreal object's already have a path. Could we rename this functionality to `IsHierarchyLocked` ?

Reply:
Maybe keep it releated to Actors and have `IsRelatedActorLocked` or even `IsActorGroupLocked` ?

Reply:
Our implementation has ultimately changed such that we don't have need of an internal function for whether something is in a locked hierarchy. This is because after Mike's suggestion, we ignore hierarchy path Actors and only care about explicitly locked Actors or locked hierarchy root Actors. So instead, `IsLocked` just checks, for both the Actor argument and Actor hierarchy root whether `IsExplicitlyLocked` or `IsLockedHierarchyRoot` returns true.